src/sntp.rs

```diff
@@ -182,6 +217,7 @@ impl Drop for EspSntp {
             let mut taken = TAKEN.lock();
 
             unsafe { sntp_stop() };
+            self.unsubscribe();
```

Comment:
Protect the call to `unsubscribe` with `#[cfg(feature = "alloc")]`.

The `alloc` changes make sure that:
* You are using Box from the `alloc` module in case user has selected `no_std` but has still enabled `alloc`.  The STD `Box` is a type-alias for `alloc::boxed::Box`, but this type alias is not available in `no_std`.
* The SNTP module can still work, but without a callback in case user has opt out even from `alloc`.

Reply:
Thanks for the explanation!

I also added it here, as SYNC_CB shouldn't be available without `alloc`.
```rs
        #[cfg(feature = "alloc")]
        if let Some(cb) = &mut *SYNC_CB.lock().unwrap() {
```